### PR TITLE
ref: Storages are CDC aware

### PR DESCRIPTION
**Pull request by lynnagara**
This moves the CDC information into the storage (rather than the
dataset). Since CDC loads data directly to a table, it seems unnecessary
to go through dataset to achieve this.

The other motivation for this is it will then allow consumers to be run
on storages rather than datasets as well.

### CODE REVIEW

File: snuba/datasets/storage.py [hunk 1]
```diff
@@ -106,6 +106,21 @@ def get_table_writer(self) -> TableWriter:
         return self.__table_writer
 
 
+class CdcStorage(WritableTableStorage):
```
Comment (fpacifici) [inline snuba/datasets/storage.py:109]: I owuld keep the CdcStorage out of here and in the datasets/cdc module. Everything cdc is there regarding the dataset model. I would argue the cdc concept is not part of the Snuba general dataset api.

File: tests/stateful_consumer/test_bootstrap_state.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def test_snapshot_loaded(self, create_consumer) -> None:
         create_consumer.return_value = kafka_consumer
 
         bootstrap = BootstrapState(
-            "cdc_control", "somewhere", "something", get_dataset("groupedmessage"),
+            "cdc_control", "somewhere", "something", get_storage("groupedmessages"),
```
Comment (fpacifici) [inline tests/stateful_consumer/test_bootstrap_state.py:117]: This is not going to type check. `get_storage` returns a Storage while the BootstrapState constructor accepts a CdcStorage. While we are not as strict with tests as we are with production code, this is an easy fix.

**Review by fpacifici — CHANGES_REQUESTED** [review]
I am ok with the design. There a re a couple of implementation changes.

File: tests/stateful_consumer/test_bootstrap_state.py (hunk 2 above)
Comment (lynnagara) [reply tests/stateful_consumer/test_bootstrap_state.py]: Oops, thanks

File: snuba/datasets/storage.py (hunk 1 above)
Comment (lynnagara) [reply snuba/datasets/storage.py]: Moved it into the CDC module

**Review by fpacifici — APPROVED** [review]

**Pull request merged**